metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/InfluxDbReporter.java

```diff
@@ -212,29 +204,39 @@ public void report(
         final SortedMap<String, Counter> counters,
         final SortedMap<String, Histogram> histograms,
         final SortedMap<String, Meter> meters,
-        final SortedMap<String, Timer> timers) {
+        final SortedMap<String, Timer> timers
+    ) {
         final long now = System.currentTimeMillis();
 
         try {
             influxDb.flush();
 
             reportGauges(gauges, now);
 
-            for (Map.Entry<String, Counter> entry : counters.entrySet()) {
-                reportCounter(entry.getKey(), entry.getValue(), now);
-            }
+            counters.forEach((key, counter) -> {
+                parse(key, (name, tags) -> reportCounter(name, counter, now, tags));
+            });
 
-            for (Map.Entry<String, Histogram> entry : histograms.entrySet()) {
-                reportHistogram(entry.getKey(), entry.getValue(), now);
-            }
+            histograms.forEach((key, histogram) -> {
+                if (canSkipMetric(key, histogram)) {
```

Comment:
with these kinds of changes we are making the repo java8 dependent, has there been a decision before to not make it java8 or do we even have other parts of the code that make J8 a must?